service-base/src/main/java/org/eclipse/hono/service/HealthCheckServer.java

```diff
@@ -0,0 +1,160 @@
+import org.eclipse.hono.util.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.core.http.HttpServer;
+import io.vertx.core.http.HttpServerOptions;
+import io.vertx.ext.healthchecks.HealthCheckHandler;
+import io.vertx.ext.web.Router;
+
+/**
+ * Provides a HTTP server for health checks. Requires an instance of {@link ApplicationConfigProperties} for it's
+ * configuration.
+ * <p>
+ * If {@code healthCheckPort} is not defined in the configuration, this implementation does nothing.
+ * <p>
+ * <b>Usage</b>
+ * <ol>
+ * <li>Invoke {@link #registerHealthCheckResources(Object)} with an instance of {@link AbstractServiceBase} to register
+ * the readiness and liveness checks defined there.</li>
+ * <li>Invoke {@link #startHealthCheckServer()} to start the server</li>
+ * <li>Before shutdown: invoke {@link #stopHealthCheckServer()} for a graceful shutdown.</li>
+ * </ol>
+ */
+public class HealthCheckServer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(HealthCheckServer.class);
+
+    private static final String URI_LIVENESS_PROBE  = "/liveness";
+    private static final String URI_READINESS_PROBE = "/readiness";
+
+    private HttpServer server;
+
+    private HealthCheckHandler readinessHandler;
+    private HealthCheckHandler livenessHandler;
+
+    private Vertx vertx;
+    private ApplicationConfigProperties config;
+    private Router router;
+
+    /**
+     * Create a new HealthCheckServer for the given Vertx and configuration.
+     *
+     * @param vertx The vertx instance.
+     * @param config The application configuration instance.
+     * @throws NullPointerException if vertx is {@code null}.
+     * @throws NullPointerException if config is {@code null}.
+     */
+    public HealthCheckServer(Vertx vertx, ApplicationConfigProperties config) {
+        this.vertx = Objects.requireNonNull(vertx);
+        this.config = Objects.requireNonNull(config);
+
+        prepareHealthCheck();
+    }
+
+    /**
+     *
+     * Configures router and healthCheckHandlers.
+     *
+     */
+    private void prepareHealthCheck() {
+        if (config.getHealthCheckPort() != Constants.PORT_UNCONFIGURED) {
+            readinessHandler = HealthCheckHandler.create(vertx);
+            livenessHandler = HealthCheckHandler.create(vertx);
+            router = Router.router(vertx);
+            LOG.debug("Health check prepared.");
+        } else {
+            LOG.info("No health check configured.");
+        }
+    }
+
+    /**
+     * Registers the readiness and liveness checks of the given service if health check is configured, otherwise does
+     * nothing.
+     * 
+     * @param serviceInstance instance of the service which's checks should be registered.
+     */
+    // TODO: change parameter to AbstractServiceBase and remove casts (see: https://github.com/eclipse/hono/pull/287).
```

Comment:
I would rather have a dedicated interface like `HealthCheckProvider` which has methods *registerLivenessChecks* and *registerReadinessChecks*. This could then be implemented by services and endpoints and whatever ... would also make testing easier, I guess.